src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -37,7 +37,7 @@
     [
         # In snowflake, a double single quote resolves as a single quote in the string.
         # https://docs.snowflake.com/en/sql-reference/data-types-text.html#single-quoted-string-constants
-        RegexLexer("single_quote", r"'([^']|'')*'", CodeSegment),
+        RegexLexer("single_quote", r"'([^'\\]|\\.|'')*'", CodeSegment),
```

Comment:
Reading regexes is hard. 😛  Is this changing single-quoted strings to allow a backlash followed by any other character?

Reply:
In effect, yes. What the middle bit says is actually:
- Anything which isn't a `'` or a `\`.
- or a `\` followed by any single character.
- or a `''`.

The reason I've made the second entry quite broad, is to cover the case that someone enters an escape sequence but not a standard one (e.g. `\q`). In this case the current regex would match that and assume it was part of the string (the desired behaviour). If however we were more specific there, then the second case wouldn't catch it and the string literal would fail to parse entirely.

What we're trying to do here is parse the string literal, not accurately capture escape sequences - if that makes sense?